protoflo/components/__init__.py

```diff
@@ -255,31 +264,28 @@ def getCache ():
 	results = []
 
 	for moduleObj in getSearchDirectories():
-
 		componentPath = moduleObj.filePath
 		dropinPath = componentPath.parent().child('components.cache')
 
-		try:
-			lastModified = componentPath.getModificationTime()
-		except:
-			log.err("Could not stat {:s}".format(str(componentPath)))
-			continue
-
 		# Look for cache
 		try:
 			lastCached = dropinPath.getModificationTime()
 			collection = pickle.load(dropinPath.open('r'))
+		# FIXME: what kind of error do we expect?
 		except:
-			lastCached = 0
-
-		
-		if lastCached < lastModified:
 			stale = True
 		else:
 			stale = False
-			for component in collection.components:
-				if FilePath(component.fileName).getModificationTime() > lastCached:
-					stale = True
+			for path in componentPath.parent().walk():
```

Comment:
I can see that the original implementation might miss files where a modification has occurred in a file that isn't the one containing the __components__ dict, but I don't think that checking all of the sibling files is the best answer. Mainly because the intention was that a component library doesn't have to be a python module, and a file such as protoflo_mycomponents.py on the Python path and containing a __components__ dict would also work (to save having lots of modules).

The answer is probably either to a) somehow keep a list of all the dependent files in the component cache, or b) enforce that a component library must be a module (though I think this would be a shame).

Any other suggestions?